api/controllers/DashboardController.js

```diff
@@ -144,48 +144,28 @@ const dashboardController = {
                 sails.log.error('No results found.');
             }
         }
-        const apiResponse = await dashboardController._getDataFromCasebook(
-            req,
-            results
-        );
+        const { data: apiResponse } =
+            await dashboardController._getDataFromCasebook(results);
+
         return dashboardController._addCasebookStatusesToResults(apiResponse, {
             ...displayAppsArgs,
             results,
         });
     },
 
-    _getDataFromCasebook(req, results) {
-        const applicationReferences = results.map(
-            (resultItem) => resultItem.unique_app_id
-        );
-
-        const queryParamsObj = {
-            timestamp: Date.now().toString(),
-            applicationReference: applicationReferences,
-        };
+    async _getDataFromCasebook(results) {
+        try {
+            const applicationReferences = results.map(
+                (resultItem) => resultItem.unique_app_id
+            );
 
-        return CasebookService.get({
-            uri: req._sails.config.customURLs.applicationStatusAPIURL,
-            json: true,
-            useQuerystring: true,
-            promise: true,
-            qs: queryParamsObj,
-        })
-            .then((response) => {
-                const responseHasErrors = response.hasOwnProperty('errors');
-                if (responseHasErrors) {
-                    sails.log.error(
-                        `Invalid response from Casebook Status API call:  ${response.message}`
-                    );
-                    return response.status(500);
-                }
-                return response;
-            })
-            .catch((err) => {
-                sails.log.error(
-                    `Error returned from Casebook API call: ${err}`
-                );
-            });
+            return await CasebookService.getApplicationStatus(
+                applicationReferences
```

Comment:
nice! I would probably be lazy and just do the .map inside the function (but you definitely should not follow that example cos it tends to get quite unreadable 😄)

Reply:
Nah I'm sure there's a way to do it and have readable code.